alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -277,7 +275,25 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
                                         responseReader
                                     ) {
                                         override fun handleException(e: TransportException) {
-                                            listener.onFailure(e)
+                                            // retry in local node
+                                            transportService.sendRequest(
```

Comment:
can we add a check to retry only if it is either NodeDisconnected or CircuitBreakerException? 
plz ensure failure from local node itself is not re-tried else we will be stuck in infinite loop

Reply:
you can refer my impl in https://github.com/opensearch-project/asynchronous-search/blob/6a00f98f1dcc462021ad43bf41c4269df54a7822/src/main/java/org/opensearch/search/asynchronous/transport/TransportAsynchronousSearchRoutingAction.java#L12

Reply:
done